lib/apis/v3/bgpconfig.go

```diff
@@ -41,18 +41,29 @@ type BGPConfiguration struct {
 type BGPConfigurationSpec struct {
 	// LogSeverityScreen is the log severity above which logs are sent to the stdout. [Default: INFO]
 	LogSeverityScreen string `json:"logSeverityScreen,omitempty" validate:"omitempty,logLevel" confignamev1:"loglevel"`
+
 	// NodeToNodeMeshEnabled sets whether full node to node BGP mesh is enabled. [Default: true]
 	NodeToNodeMeshEnabled *bool `json:"nodeToNodeMeshEnabled,omitempty" validate:"omitempty" confignamev1:"node_mesh"`
+
 	// ASNumber is the default AS number used by a node. [Default: 64512]
 	ASNumber *numorstring.ASNumber `json:"asNumber,omitempty" validate:"omitempty" confignamev1:"as_num"`
-	// ServiceExternalIPs are the CIDR blocks for service Kubernetes External IPs.
-	// Kubernetes Service ExternalIPs will only be advertised if they are within
-	// one of these blocks.
-	ServiceExternalIPs []SvcExternalIPBlock `json:"serviceExternalIPs,omitempty" validate:"omitempty" confignamev1:"svc_external_ips"`
+
+	// ServiceExternalIPs are the CIDR blocks for Kubernetes Service External IPs.
+	// Kubernetes Service ExternalIPs will only be advertised if they are within one of these blocks.
+	ServiceExternalIPs []ServiceExternalIPBlock `json:"serviceExternalIPs,omitempty" validate:"omitempty" confignamev1:"svc_external_ips"`
+
+	// ServiceClusterIPs are the CIDR blocks from which service cluster IPs are allocated.
+	// If specified, Calico will advertise these blocks, as well as any cluster IPs within them.
+	ServiceClusterIPs []ServiceClusterIPBlock `json:"serviceClusterIPs,omitempty" validate:"omitempty" confignamev1:"svc_cluster_ips"`
+}
+
+// ServiceExternalIPBlock represents a single whitelisted CIDR External IP block.
+type ServiceExternalIPBlock struct {
+	CIDR string `json:"cidr,omitempty" validate:"omitempty,net"`
 }
 
-// SvcExternalIPBlock represents a single whitelisted CIDR External IP block.
-type SvcExternalIPBlock struct {
+// ServiceClusterIPBlock represents a single whitelisted CIDR block for ClusterIPs.
+type ServiceClusterIPBlock struct {
```

Comment:
I don't want to block this PR, but I'm curious what the benefit of ServiceClusterIPBlock is, instead of the more direct:
```
	ServiceClusterIPs []string
```
With the extra indirection it feels like the construction of these resources will be more awkward in both YAML and golang.

Reply:
@neiljerram good question - the reason we designed it this way is for extensibility down the line. This model allows us to augment blocks with additional attributes if needed, and the yaml difference is minor.

The difference now is this:

```
serviceClusterIPs:
- cidr: 10.0.0.0/8
```

vs 

```
serviceClusterIPs:
- 10.0.0.0/8
```

But, the former allows us to add additional fields if we want to in the future, whereas the second form doesn't allow us to easily do that. For example:

```
serviceClusterIPs:
- cidr: 10.0.0.0/8
  nodeSelector: 'all()'
```

Reply:
In this case I struggle to imagine what other extensions there could be; but OK.